spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/MqttPahoMessageHandler.java

```diff
@@ -208,7 +204,7 @@ private void sendDeliveryComplete(IMqttDeliveryToken token) {
 	}
 
 	@Override
-	public void connectionLost(Throwable cause) {
+	public synchronized void connectionLost(Throwable cause) {
```

Comment:
OK. I think I apply it having the `client.setCallback(this);` fact. 
So, any concurrent `synchronized` method invocation on this object causes a blocking.
Correct ?
Just because I'm sure that this `connectionLost()` is called from the different Thread.

Although I always thought that we are blocking only on the single method, but not on different...